doublyLinkedLists.js

```diff
@@ -0,0 +1,499 @@
+                return this.deleteFromFront(position);
+                
+            } else {
+                
+                // if position is tail, set tail equal to newNode
+                if (position === this._size) {
+                    return this.deleteLast();
+                }
+                
+                // delete node by crawling from end
+                return this.deleteFromEnd(position);
+                
+            }
+            
+        }
+        
+        throw new Error('Position out of bounds');
+        
+    },
+    
+    // delete first node found containing the specified data
+    delete: function (data) {
+        
+        var current = this.head,
+            next,
+            previous;
+        
+        if (this._size > 0) {
+        
+            // if head contains data, delete head
+            if (data === current.data) {
+
+                this.head = current.next;
+                this._size--;
+
+                return true;
+
+            } else {
+
+                // crawl thorugh list until data is found and delete the node cointaining it
+                for (var i = 1; i < this._size - 1; i++) {
+
+                    previous = current;
+                    current = current.next;
+                    next = current.next;
+
+                    if (data === current.data) {
+
+                        previous.next = next;
+                        next.previous = previous;
+                        this._size--;
+
+                        return true;
+                    }
+                }
+            }
+        }
+        
+        // if node isn't throw an error
+        throw new Error('The specified node was not found');
+        
+    },
+    
+    // logs the full list from head to tail
+    displayForward: function () {
+        
+        var current = this.head,
+            string = '';
+        
+        // crawl through every nodes, appending each data to string
+        while (current) {
+            
+            string += (current.data + ',');
+            current = current.next;
+            
+        }
+        
+        console.log(string);
+        
+    },
+    
+    // logs the full list from tail to head
+    displayBackward: function () {
+        
+        var current = this.tail,
+            string = '';
+        
+        // crawl through every nodes, appending each data to string
+        while (current) {
+            
+            string += (current.data + ',');
+            current = current.previous;
+            
+        }
+        
+        console.log(string);
+        
+    },
+    
+    // get size of list
+    getSize: function (){
+        
+        console.log(this._size);
+        return this._size;
+        
+    },
+    
+    // delete all elements from list
+    clear: function (){
+        this.head = this.tail = null;
+        this._size = 0;
+    }
+    
+};
+
+var doublyLinkedList = new DoublyLinkedList();
```

Comment:
need to write proper unit tests, I'm not sure what's the best way in javascript but you could just make an object with a bunch of test methods and call them in a for loop.
Each method should test one very specific feature of the code. Typically for a data structure like this you would end up  with a lot more code in the tests than in the data structure.
An example would be:
```
deleteDataAtFirstIndexTest() {
dll = new dll()
dll.insertFirst(3)
dll.insertFirst(2)
dll.insertFirst(1)
var r = dll.deleteData(1)
assert dll.getSize() == 0
assert r == 1
assert dll.head == 2
assert dll.tail == 3
```
I will also make you want to limit the size of your API :D